ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml

```diff
@@ -76,13 +76,20 @@
 
 - name: Remove c:\cygwin64\bin from the path if it exists
   win_path:
+    name: PATH
     elements:
       - 'C:\cygwin64\bin'
+    scope: machine
     state: absent
   tags: cygwin
 
-- name: Add c:\cygwin64\bin to the start of the %PATH%
-  raw: setx PATH "C:\cygwin64\bin;$env:Path" /m
+- name: Add c:\cygwin64\bin to %PATH%
```

Comment:
setx is needed to make sure cygwin is always on the front of the path.  If other tools install/write to the front of the path, like openssh, cygwin won't work as desired.

Reply:
Using the `win_shell` module with `cmd` allows the setx command to work without causing errors on playbook reruns.